test/fixtures/rules/std_rule_cases/L036.yml

```diff
@@ -153,7 +151,59 @@ test_distinct_single_fail_b:
     SELECT
       distinct a
     FROM my_table
-  # NOTE: This doesn't yet reflow nicely. We should solve that elsewhere.
   fix_str: |
     SELECT distinct a
-       FROM my_table
+    FROM my_table
+
+test_long_single_select_clause:
+  fail_str: "SELECT\n100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\n"
+
+test_single_select_with_comment_after_column:
+  fail_str: |
+    SELECT
+      1 -- this is a comment
+    FROM
+      my_table
+  fix_str: |
+    SELECT 1
+      -- this is a comment
```

Comment:
Possibly. And went through an iteration like that but it gets tricky with other comments. Think about this:

```sql
SELECT
     -- We should return 1 when the value is weird
     1 -- like this!
```

It would become this:

```sql
SELECT 1 -- like this!
     -- We should return 1 when the value is weird
```

Also moving a comment up leads to a greater likelihood of clashing with L016 if line length grows.

As this only promotion up to the `SELECT` line only happens for singular lines I don'm think it's too bad to have this, as the comment is still close enough to the column:

```sql
    SELECT 1
      -- this is a comment
    FROM
      my_table
```

So I'd prefer to keep as is.

Reply:
Sounds good. I think this came up in the past as well. Happy to delay dealing with messy stuff -- progress on fixing bugs is more important than these minor formatting details. 👍🏽